waifuc/source/pixiv.py

```diff
@@ -168,21 +168,22 @@ def _iter_data(self) -> Iterator[Tuple[Union[str, int], str, dict]]:
                                  f'in urls {zip_urls!r}, skipped.')
                     continue
 
-                try:
-                    gif_image = self._make_gif_for_ugoira(frame_infos, zip_url)
-                    gif_image.load()
-                except _UgoiraSkip:
-                    continue
-                filename = f'{self.group_name}_{illust["id"]}.gif'
-                meta = {
-                    'pixiv': illust,
-                    'ugoira': metadata,
-                    'group_id': f'{self.group_name}_{illust["id"]}',
-                    'instance_id': f'{self.group_name}_{illust["id"]}',
-                    'filename': filename,
-                    'url': zip_url,
-                }
-                yield f'{illust["id"]}', gif_image, meta
+                if self.download_gif:
```

Comment:
i think maybe it's better to give use a selection, like:
1. download gif image
2. dont download gif, but download the seperated images, and rename each of them
3. just ignore all the ugoiras